firebase-perf/src/main/java/com/google/firebase/perf/session/PerfSession.java

```diff
@@ -23,62 +23,53 @@
 import com.google.firebase.perf.util.Clock;
 import com.google.firebase.perf.util.Timer;
 import com.google.firebase.perf.v1.SessionVerbosity;
-import com.google.firebase.sessions.api.SessionSubscriber;
 import java.util.List;
+import java.util.UUID;
 import java.util.concurrent.TimeUnit;
 
 /** Details of a session including a unique Id and related information. */
 public class PerfSession implements Parcelable {
-
-  private final String sessionId;
   private final Timer creationTime;
-  @Nullable private String aqsSessionId;
-
+  private final String sessionId;
   private boolean isGaugeAndEventCollectionEnabled = false;
+  public final boolean isAqsReady;
 
   /*
    * Creates a PerfSession object and decides what metrics to collect.
    */
-  public static PerfSession createWithId(@NonNull String sessionId) {
-    String prunedSessionId = sessionId.replace("-", "");
-    PerfSession session = new PerfSession(prunedSessionId, new Clock());
-    session.setGaugeAndEventCollectionEnabled(shouldCollectGaugesAndEvents());
-
+  public static PerfSession createWithId(@Nullable String aqsSessionId) {
+    String sessionId = UUID.randomUUID().toString().replace("-", "");
```

Comment:
Can you refactor this to an if-else so we don't generate a random uuid and then not use it?

Also boolean not Boolean for primitive type

Reply:
sg, also fyi the change currently breaks a bunch of unit tests, didn't expect that so I gonna take a further look

Reply:
done